server/aesdsocket.c

```diff
@@ -99,6 +142,199 @@ static int daemon_create()
+            char *temp_ptr  = (char *)realloc(storage_buffer, (storage_buffer_size + INITIAL_BUFFER_SIZE));
+            if (temp_ptr) {
+                storage_buffer = temp_ptr;
+                storage_buffer_size += INITIAL_BUFFER_SIZE;
+                storage_buffer_size_count += 1;
+            } else {
+                printf("Unable to allocate memory\n");
+            }
+        }
+
+        int i;
+        // Iterating over bytes received to check for "\n" character
+        for (i = 0; i < bytes_received; i++) {
+            if (buffer[i] == '\n') {
+                packet_complete_flag = 1;
+                temp_iterator = i+1;
+                break;
+            }
+        }
+
+        // Copying from receive buffer to storage buffer 
+        memcpy(storage_buffer + (storage_buffer_size_count - 1)*INITIAL_BUFFER_SIZE + previous_byte_count, buffer, bytes_received);
+
+        // Check if data packet is complete and assign bytes to write based on temporary iterator
+        if (packet_complete_flag == 1) {
+            bytes_to_write = (storage_buffer_size_count - 1)*INITIAL_BUFFER_SIZE + temp_iterator + previous_byte_count;
+            previous_byte_count = 0;
+            break;
+        } else {
+            // Recheck if \n not received using i and verifying if bytes recerive is less than buffer size
+            if ((i == bytes_received) && (bytes_received < INITIAL_BUFFER_SIZE)) {
+                previous_byte_count += bytes_received;
+            }
+
+            else if (bytes_received == INITIAL_BUFFER_SIZE){
+                // Increasing size of storage buffer by INITIAL_BUFFER_SIZE amount if no "\n" character found in buffer
+                char *temp_ptr  = (char *)realloc(storage_buffer, (storage_buffer_size + INITIAL_BUFFER_SIZE));
+                if (temp_ptr) {
+                    storage_buffer = temp_ptr;
+                    storage_buffer_size += INITIAL_BUFFER_SIZE;
+                    storage_buffer_size_count += 1;
+                } else {
+                    printf("Unable to allocate memory\n");
+                }
+            }
+        }
+    }
+
+    if (bytes_received == -1) {
+        syslog(LOG_ERR, "Error in receive; errno is %d\n", errno);
+        goto receive_error;
+    }
+
+    // Check if packet_complete; indicating "\n" encountered in received data
+    if (packet_complete_flag == 1) {
+
+        packet_complete_flag = 0;
+
+        int ret = pthread_mutex_lock(&mutex);
+        if (ret != 0) {
+            perror("Mutex Lock");
+        }
+        //Writing to file
+        int nr = write(testfile_fd, storage_buffer, bytes_to_write);
+        file_size += nr; // Adding to current file length
+
+        lseek(testfile_fd, 0, SEEK_SET); // Setting the FD to start of file
+
+        // Buffered read and send operation for 
+        char *read_buffer = NULL;
+
+        int read_buffer_size;
+
+        if (file_size < WRITE_BUFFER_SIZE) {
```

Comment:
@GautamaGandhi nice change from assignment 5... however I don't think you need this check here and I don't think you necessarily need dynamic allocation for this buffer.  You could always just use a fixed size and if file_size was smaller it would be reflected in `bytes_read` returned from the read function.

Reply:
@dwalkes I did this check to assign the size of the malloced block to be capped off at 512 bytes. You are right, a static fixed size buffer would do this as well. I will make changes to remove the dynamic allocation component.